src/release/breaking-changes/shortcut-key-event-migration.md

```diff
@@ -0,0 +1,173 @@
+already do this.
+
+The change also modifies the `ShortcutManager.handleKeypress` method to take a
+`KeyEvent` instead of `RawKeyEvent`.
+
+## Migration guide
+
+APIs provided by the Flutter framework are already migrated. Migration is
+needed only if you're using any of the methods listed in the previous section.
+
+### Migrating your APIs that use `ShortcutActivator` or its subclasses.
+
+Pass a `KeyEvent` instead of a `RawKeyEvent` to `ShortcutActivator.accepts`.
+This may mean switching where you get your key events from. Depending on where
+you get them, this can either mean switching to using `Focus.onKeyEvent` instead
+of `Focus.onKey`, or a similar change if using `FocusScope`, `FocusNode` or
+`FocusScopeNode`.
+
+If you're using a `RawKeyboardListener`, switch to using a
+`KeyboardListener` instead. If you're accessing `RawKeyboard` directly, use
+`HardwareKeyboard` instead. You'll find that there are non-raw equivalents for
+all of the key event sources.
+
+### Migrating your APIs that extend `ShortcutActivator`
+
+The `ShortcutActivator.accepts` method was modified to take a `KeyEvent` and a
+`HardwareKeyboard` instead of a `RawKeyEvent` and `RawKeyboard`.
+
+Before:
+
+```dart
+class MyActivator extends ShortcutActivator {
+  @override
+  bool accepts(RawKeyEvent event, RawKeyboard state) {
+    // ... (your implementation here)
+    returns false;
+  }
+  // ...
+}
+```
+
+After:
+
+```dart
+class MyActivator extends ShortcutActivator {
+  @override
+  bool accepts(KeyEvent event, HardwareKeyboard state) {
+    // ... (your implementation here)
+    returns false;
+  }
+  // ...
+}
+```
+
+### Migrating your APIs that extend `ShortcutManager`
+
+The `ShortcutManager` class was modified to take `KeyEvent`s in `handleKeypress`
+instead of `RawKeyEvent`s.  One difference in the two APIs is that repeated keys
+are determined differently. In the `RawKeyEvent` case, the `repeat` member
+indicated a repeat, but in `RawKeyEvent` code, the event is a different type
+(`KeyRepeatEvent`).
+
+Before:
+
+```dart
+class _MyShortcutManager extends ShortcutManager {
+  @override
+  KeyEventResult handleKeypress(BuildContext context, RawKeyEvent event) {
+    if (event is! RawKeyDownEvent) {
+      return KeyEventResult.ignored;
+    }
+    if (event.repeat) {
+      // (Do something with repeated keys.)
+    }
+    // ... (your implementation here)
+    return KeyEventResult.handled;
+  }
+}
+```
+
+After:
+
+```dart
+class _MyShortcutManager extends ShortcutManager {
+  @override
+  KeyEventResult handleKeypress(BuildContext context, KeyEvent event) {
+    if (event is! KeyDownEvent && event is! KeyRepeatEvent) {
+      return KeyEventResult.ignored;
+    }
+    if (event is KeyRepeatEvent) {
+      // (Do something with repeated keys.)
+    }
+    // ... (your implementation here)
+    return KeyEventResult.handled;
+  }
+}
+```
+
+## Timeline
+
+Landed in version: 3.17.0-5.0.pre<br>
+In stable release: not yet (Not in 3.17)
```

Comment:
I'm guessing this is targeting 3.19ish?

Reply:
No need to add a specific release, we'll make sure it gets updated once its stable.